include/openthread/instance.h

```diff
@@ -254,6 +254,62 @@ OTAPI otError OTCALL otSetStateChangedCallback(otInstance *aInstance, otStateCha
 OTAPI void OTCALL otRemoveStateChangeCallback(otInstance *aInstance, otStateChangedCallback aCallback,
                                               void *aCallbackContext);
 
+
+/**
+ * This function pointer is called to notify that a child is being added or removed from child table.
+ *
+ * @param[in]  aChildInfo  A pointer to child information structure.
+ *
+ */
+typedef void (*otChildUpdatedCallback)(const otChildInfo *aChildInfo);
+
+/**
+ * This function gets the "child added" callback function.
+ *
+ * @returns  The callback function pointer.
+ *
+ */
+otChildUpdatedCallback otGetChildAddedCallback(otInstance *aInstance);
+
+/**
+ * This function sets the "child added" callback handler.
+ *
+ * The provided callback (if non-NULL) will be invoked when a child entry is being added to the child table.
+ * Subsequent calls to this method will overwrite the previous callback. Note that this callback in emitted while
+ * the child table is being updated and always before the `otStateChangedCallback`.
+ *
+ * @param[in] aCallback                  A pointer to callback handler function.
+ *
+ * @retval OT_ERROR_NONE                 The callback was updated successfully.
+ * @retval OT_ERROR_DISABLED_FEATURE     The feature is disabled (MTD build).
+ *
+ */
+otError otSetChildAddedCallback(otInstance *aInstance, otChildUpdatedCallback aCallback);
+
+/**
+ * This function gets the "child removed" callback function.
+ *
+ * @returns  The callback function pointer.
+ *
+ */
+otChildUpdatedCallback otGetChildRemovedCallback(otInstance *aInstance);
+
+/**
+ * This function sets the "child removed" callback handler.
+ *
+ * The provided callback (if non-NULL) will be invoked when a child entry is being removed from the child table.
+ * Subsequent calls to this method will overwrite the previous callback. Note that this callback in emitted while
+ * the child table is being updated and always before the `otStateChangedCallback`.
+ *
+ * @param[in] aCallback                  A pointer to callback handler function.
+ *
+ * @retval OT_ERROR_NONE                 The callback was updated successfully.
+ * @retval OT_ERROR_DISABLED_FEATURE     The feature is disabled (MTD build).
+ *
+ */
+
+otError otSetChildRemovedCallback(otInstance *aInstance, otChildUpdatedCallback aCallback);
```

Comment:
Is it valuable to have separate callbacks for each child event?  Should we have a single callback and pass an enum or flags instead to indicate the event(s)?

Reply:
Good idea! Will change it... it will simplify the API.

Reply:
Combined the two callbacks into one in the new pushed commit.